### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by NejcZdovc**
core implementation: https://github.com/brave/brave-core/pull/950

### CODE REVIEW

File: src/ledger_impl.cc [hunk 1]
```diff
@@ -189,11 +191,20 @@ void LedgerImpl::OnLedgerStateLoaded(ledger::Result result,
                                         const std::string& data) {
   if (result == ledger::Result::LEDGER_OK) {
     if (!bat_state_->LoadState(data)) {
+      LOG(ledger_client_, ledger::LogLevel::LOG_ERROR) <<
+        "Successfully loaded but failed to parse ledger state: " << data;
+
       OnWalletInitialized(ledger::Result::INVALID_LEDGER_STATE);
     } else {
+      LOG(ledger_client_, ledger::LogLevel::LOG_INFO) <<
```
Comment (NejcZdovc) [inline src/ledger_impl.cc]: I think this is not needed

File: src/ledger_impl.cc [hunk 2]
```diff
@@ -206,8 +217,17 @@ void LedgerImpl::OnPublisherStateLoaded(ledger::Result result,
                                         const std::string& data) {
   if (result == ledger::Result::LEDGER_OK) {
     if (!bat_publishers_->loadState(data)) {
+      LOG(ledger_client_, ledger::LogLevel::LOG_ERROR) <<
+        "Successfully loaded but failed to parse ledger state: " << data;
+
       result = ledger::Result::INVALID_PUBLISHER_STATE;
+    } else {
+      LOG(ledger_client_, ledger::LogLevel::LOG_INFO) <<
```
Comment (NejcZdovc) [inline src/ledger_impl.cc]: I think this is not needed

File: src/ledger_impl.cc [hunk 3]
```diff
@@ -234,7 +254,16 @@ void LedgerImpl::LoadPublisherList(ledger::LedgerCallbackHandler* handler) {
 void LedgerImpl::OnPublisherListLoaded(ledger::Result result,
                                        const std::string& data) {
   if (result == ledger::Result::LEDGER_OK) {
-    bat_publishers_->loadPublisherList(data);
+    if (!bat_publishers_->loadPublisherList(data)) {
+      LOG(ledger_client_, ledger::LogLevel::LOG_ERROR) <<
+        "Successfully loaded but failed to parse publish list: " << data;
+    } else {
+      LOG(ledger_client_, ledger::LogLevel::LOG_INFO) <<
+        "Successfully loaded publisher list: " << data;
```
Comment (NejcZdovc) [inline src/ledger_impl.cc]: I think this is not needed

File: src/ledger_impl.cc [hunk 4]
```diff
@@ -505,7 +539,7 @@ void LedgerImpl::VotePublishers(const std::vector<braveledger_bat_helper::WINNER
 void LedgerImpl::PrepareVoteBatchTimer() {
   uint64_t start_timer_in = braveledger_bat_helper::getRandomValue(10, 60);
 
-  LOG(ledger_client_, ledger::LogLevel::LOG_ERROR) <<
+  LOG(ledger_client_, ledger::LogLevel::LOG_INFO) <<
```
Comment (NejcZdovc) [inline src/ledger_impl.cc]: I would move this to debug

File: src/ledger_impl.cc [hunk 5]
```diff
@@ -514,7 +548,7 @@ void LedgerImpl::PrepareVoteBatchTimer() {
 void LedgerImpl::VoteBatchTimer() {
   uint64_t start_timer_in = braveledger_bat_helper::getRandomValue(10, 60);
 
-  LOG(ledger_client_, ledger::LogLevel::LOG_ERROR) <<
+  LOG(ledger_client_, ledger::LogLevel::LOG_INFO) <<
```
Comment (NejcZdovc) [inline src/ledger_impl.cc]: I would move this to debug

File: src/url_request_handler.cc [hunk 6]
```diff
@@ -24,7 +24,7 @@ void URLRequestHandler::OnURLRequestResponse(uint64_t request_id,
                                             const std::string& response,
                                             const std::map<std::string, std::string>& headers) {
   if (!RunRequestHandler(request_id, response_code == 200, response, headers)) {
-    LOG(ERROR) << "no request handler found for " << request_id;
+    // LOG(ERROR) << "no request handler found for " << request_id;
```
Comment (NejcZdovc) [inline src/url_request_handler.cc]: why was this commented out?

File: src/bat_client.cc [hunk 7]
```diff
@@ -299,7 +303,8 @@ void BatClient::reconcile(const std::string& viewingId,
     }
 
     if (fee + ac_amount > balance) {
-      LOG(ledger_, ledger::LogLevel::LOG_ERROR) << "You do not have enough funds to do recurring and auto contribution";
+      LOG(ledger_, ledger::LogLevel::LOG_INFO) <<
```
[the remark this answers is not in the document]
Comment (tmancey) [reply src/bat_client.cc]: Agreed and changed to WARNING

File: src/bat_client.cc [hunk 8]
```diff
@@ -263,11 +263,13 @@ void BatClient::reconcile(const std::string& viewingId,
...
     if (list.size() == 0 || ac_amount > balance) {
       if (list.size() == 0) {
-        LOG(ledger_, ledger::LogLevel::LOG_INFO) << "Auto contribution table is empty";
+        LOG(ledger_, ledger::LogLevel::LOG_INFO) <<
+          "Auto contribution table is empty";
       }
 
       if (ac_amount > balance) {
-        LOG(ledger_, ledger::LogLevel::LOG_INFO) << "You do not have enough funds for auto contribution";
+        LOG(ledger_, ledger::LogLevel::LOG_INFO) <<
```
[the remark this answers is not in the document]
Comment (tmancey) [reply src/bat_client.cc]: Agreed and changed to WARNING

File: src/bat_client.cc [hunk 9]
```diff
@@ -325,7 +333,8 @@ void BatClient::reconcile(const std::string& viewingId,
     }
 
     if (fee > balance) {
-      LOG(ledger_, ledger::LogLevel::LOG_ERROR) << "You do not have enough funds to do a tip";
+      LOG(ledger_, ledger::LogLevel::LOG_INFO) <<
```
[the remark this answers is not in the document]
Comment (tmancey) [reply src/bat_client.cc]: Agreed and changed to WARNING

File: src/bat_get_media.cc [hunk 10]
```diff
@@ -59,10 +59,18 @@ void BatGetMedia::processMedia(const std::map<std::string, std::string>& parts,
     return;
   }
   std::string mediaId = braveledger_bat_helper::getMediaId(parts, type);
+  LOG(ledger_, ledger::LogLevel::LOG_INFO) <<
```
[the remark this answers is not in the document]
Comment (tmancey) [reply src/bat_get_media.cc]: Added LOG_DEBUG and changed from LOG_INFO to LOG_DEBUG

File: src/bat_get_media.cc [hunk 11]
```diff
@@ -59,10 +59,18 @@ void BatGetMedia::processMedia(const std::map<std::string, std::string>& parts,
...
+    "Media request (" << type << "):";
+  for (std::pair<std::string, std::string> const& value: headers) {
+    LOG(ledger_client_, ledger::LogLevel::LOG_INFO) <<
+      ">> part " << value.first << " | " << value.second;
+  }
+  LOG(ledger_, ledger::LogLevel::LOG_INFO) << "Media Id: " << mediaId;
```
[the remark this answers is not in the document]
Comment (tmancey) [reply src/bat_get_media.cc]: Added LOG_DEBUG and changed from LOG_INFO to LOG_DEBUG

File: src/bat_get_media.cc [hunk 12]
```diff
@@ -59,10 +59,18 @@ void BatGetMedia::processMedia(const std::map<std::string, std::string>& parts,
...
   if (mediaId.empty()) {
     return;
   }
   std::string media_key = braveledger_bat_helper::getMediaKey(mediaId, type);
+  LOG(ledger_, ledger::LogLevel::LOG_INFO) << "Media key: " << media_key;
```
[the remark this answers is not in the document]
Comment (tmancey) [reply src/bat_get_media.cc]: Added LOG_DEBUG and changed from LOG_INFO to LOG_DEBUG

File: src/bat_get_media.cc [hunk 13]
```diff
@@ -77,6 +85,7 @@ void BatGetMedia::processMedia(const std::map<std::string, std::string>& parts,
       twitchEventInfo.time_ = iter->second;
     }
   }
+  LOG(ledger_, ledger::LogLevel::LOG_INFO) << "Media duration: " << duration;
```
[the remark this answers is not in the document]
Comment (tmancey) [reply src/bat_get_media.cc]: Added LOG_DEBUG and changed from LOG_INFO to LOG_DEBUG

File: src/bat_get_media.cc [hunk 14]
```diff
@@ -511,6 +527,9 @@ void BatGetMedia::onMediaActivityError(const ledger::VisitData& visit_data,
     name = TWITCH_MEDIA_TYPE;
   }
 
+  LOG(ledger_, ledger::LogLevel::LOG_ERROR) << "Media activity error for " <<
```
[the remark this answers is not in the document]
Comment (tmancey) [reply src/bat_get_media.cc]: Added LOG_DEBUG and changed from LOG_INFO to LOG_DEBUG

[the remark this answers is not in the document]
Comment (tmancey) [reply src/bat_get_media.cc]: Removed

[the remark this answers is not in the document]
Comment (tmancey) [reply src/bat_get_media.cc]: Is this covered elsewhere, it was taken from Muon's logging?

File: src/bat_publishers.cc [hunk 15]
```diff
@@ -821,9 +821,15 @@ bool BatPublishers::loadState(const std::string& data) {
 
 void BatPublishers::OnPublisherStateSaved(ledger::Result result) {
   if (result != ledger::Result::LEDGER_OK) {
-    LOG(ledger_, ledger::LogLevel::LOG_ERROR) << "Could not save publisher state";
+    LOG(ledger_, ledger::LogLevel::LOG_ERROR) <<
+      "Could not save publisher state";
     // TODO - error handling
     return;
+  } else {
+    std::string json;
```
[the remark this answers is not in the document]
Comment (tmancey) [reply src/bat_publishers.cc]: This was also taken from Muon's logging, are you sure you want me to remove?

File: src/ledger_impl.cc (hunk 1 above)
Comment (tmancey) [reply src/ledger_impl.cc]: This was also taken from Muon's logging, are you sure you want me to remove? I have added these states as LOG_INFO to ads as it helps diagnose issues quickly

File: src/ledger_impl.cc (hunk 2 above)
Comment (tmancey) [reply src/ledger_impl.cc]: This was also taken from Muon's logging, are you sure you want me to remove? I have added these states as LOG_INFO to ads as it helps diagnose issues quickly

File: src/ledger_impl.cc (hunk 3 above)
Comment (tmancey) [reply src/ledger_impl.cc]: This was also taken from Muon's logging, are you sure you want me to remove? I have added these states as LOG_INFO to ads as it helps diagnose issues quickly

File: src/ledger_impl.cc (hunk 4 above)
Comment (tmancey) [reply src/ledger_impl.cc]: Moved to debug

File: src/ledger_impl.cc (hunk 5 above)
Comment (tmancey) [reply src/ledger_impl.cc]: Moved to debug

**Comment by tmancey** [comment]
Comments addressed in latest commit

File: src/ledger_impl.cc (hunk 2 above)
Comment (NejcZdovc) [reply src/ledger_impl.cc]: yes, we don't need success

File: src/ledger_impl.cc (hunk 1 above)
Comment (NejcZdovc) [reply src/ledger_impl.cc]: yes, we don't need success

File: src/ledger_impl.cc (hunk 3 above)
Comment (NejcZdovc) [reply src/ledger_impl.cc]: yes, we don't need success

File: src/url_request_handler.cc (hunk 6 above)
Comment (NejcZdovc) [reply src/url_request_handler.cc]: ok

File: src/bat_get_media.cc (hunk 14 above)
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply src/bat_get_media.cc]: not needed

File: src/bat_publishers.cc (hunk 15 above)
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply src/bat_publishers.cc]: yes

**Pull request merged**